web/src/app/util/avatars.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useEffect, useState } from 'react'
+import { Layers, RotateRight, Today, VpnKey, Person } from '@material-ui/icons'
+import { useSessionInfo } from './RequireConfig'
+import { Avatar, AvatarTypeMap, SvgIconProps } from '@material-ui/core'
+import { OverridableComponent } from '@material-ui/core/OverridableComponent'
+import _ from 'lodash-es'
+
+type IconProps = (props: SvgIconProps) => JSX.Element
+type AvatarProps = OverridableComponent<AvatarTypeMap>
+interface UserAvatarProps extends AvatarProps {
+  userID: string
+}
+
+function useValidImage(srcURL?: string) {
+  const [valid, setValid] = useState(false)
+
+  useEffect(() => {
+    setValid(false)
+    if (!srcURL) return
+
+    const image = new Image()
+    image.onload = () => setValid(true)
+    image.src = srcURL
+
+    return () => {
+      image.onload = null
+    }
+  }, [srcURL])
+
+  return valid
+}
+
+function useAvatar(
+  Fallback: IconProps,
+  otherProps: AvatarProps | UserAvatarProps,
+  imgSrc?: string,
+) {
+  const validImage = useValidImage(imgSrc)
+
+  const DOMProps = _.omit(otherProps, 'userID')
+
+  return (
+    <Avatar
+      alt=''
+      src={validImage ? imgSrc : undefined}
+      data-cy={validImage ? null : 'avatar-fallback'}
+      {...DOMProps}
+    >
+      {validImage ? null : <Fallback fontSize='small' />}
+    </Avatar>
+  )
+}
+
+export function UserAvatar(props: UserAvatarProps) {
+  const { userID } = props
+  return useAvatar(Person, props, `/api/v2/user-avatar/${userID}`)
```

Comment:
We don't need UserAvatarProps inside `useAvatar` let's clean it up in `UserAvatar` instead.
```suggestion
  otherProps: AvatarProps,
  imgSrc?: string,
) {
  const validImage = useValidImage(imgSrc)

  return (
    <Avatar
      alt=''
      src={validImage ? imgSrc : undefined}
      data-cy={validImage ? null : 'avatar-fallback'}
      {...otherProps}
    >
      {validImage ? null : <Fallback fontSize='small' />}
    </Avatar>
  )
}

export function UserAvatar(props: UserAvatarProps) {
  const { userID, ...otherProps } = props
  return useAvatar(Person, otherProps, `/api/v2/user-avatar/${userID}`)
```

Reply:
We can cast it manually if the inferencing isn't working automatically.